marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/reader/ReadAllDataTypesTest.java

```diff
@@ -0,0 +1,91 @@
+            assertEquals(1, row.getInt(1), "int");
+            assertEquals(Byte.decode("1"), row.getByte(2),"byte");
+            assertEquals(1, row.getInt(3),"positiveInteger");
+            assertEquals(1, row.getInt(4), "nonNegativeInteger");
+            assertEquals(1L, row.getLong(5), "long");
+            assertEquals(1L, row.getLong(6), "unsignedLong");
+            assertEquals(1L, row.getLong(7), "unsignedInt");
+            assertEquals(Short.decode("1"), row.getShort(8), "short");
+            assertEquals(Short.decode("1"), row.getShort(9), "unsignedByte");
+            assertEquals(Short.decode("1"), row.getShort(10), "unsignedShort");
+            assertEquals("John", row.getString(11), "string");
+            assertEquals(6218, row.get(12, DataTypes.DateType), "date");
+            assertEquals(537336758000000L, row.get(13, DataTypes.TimestampType), "dateTime");
+            assertEquals("20:12:38", row.getString(14), "time");
+            assertEquals("/John/IRI", row.getString(15), "IRI");
+            assertEquals("/John/IRI", row.getString(16), "anyURI");
+            assertEquals(true, row.getBoolean(17), "boolean");
+            assertEquals(-10, row.getInt(18), "negativeInteger");
+            assertEquals(-10, row.getInt(19), "nonPositiveInteger");
+            assertEquals(30.5f, row.getFloat(20), "float");
+            assertEquals(30.5d, row.getDouble(21), "double");
+            assertEquals(30.5d, row.getDouble(22), "decimal");
+            assertEquals("1987-01", row.getString(23), "gYearMonth");
+            assertEquals("1987+02:00", row.getString(24), "gYear");
+            assertEquals("--01+02:00", row.getString(25), "gMonth");
+            assertEquals("---10", row.getString(26), "gDay");
+            assertEquals("--01-10", row.getString(27), "gMonthDay");
+            assertEquals("P1M", row.getString(28), "duration");
+            assertEquals("P1M", row.getString(29), "yearMonthDuration");
+            assertEquals("P30DT1H", row.getString(30), "dayTimeDuration");
+            assertEquals("37.389965,-122.07858", row.getString(31), "point");
+            assertEquals("37.389965,-122.07858", row.getString(32), "longLatPoint");
+
+            try {
+                String base64BinaryString = new String(row.getBinary(33), "UTF-8");
+                String hexBinaryString = new String(row.getBinary(34), "UTF-8");
+                assertEquals("RHVtIHNwaXJvIHNwZXJv", base64BinaryString, "base64Binary");
+                assertEquals("44756D20737069726F20737065726F", hexBinaryString, "hexBinary");
+                String s1 = new String(Base64.decode(base64BinaryString), "UTF-8");
+                String s2 = new String(Hex.decodeHex(hexBinaryString.toCharArray()), "UTF-8");
+                assertEquals("Dum spiro spero", s1);
+                assertEquals(s1, s2);
+
+            } catch (Exception e) {
+                logger.error("Failed to decode binary string: cause; " + e.getMessage());
```

Comment:
For tests, just do throw new RuntimeException(e). "fail" is really for when you expect an exception to occur, but one doesn't - in which case, it's better to use assertThrows.

Reply:
Better yet - don't catch anything, just add "throws Exception" to the test method